core/mongoinit.go

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+type MongoCredentials struct {
+	AdminUser           string `json:"admin,omitempty"`
+	AdminPasswd         string `json:"adminpasswd,omitempty"`
+	MetadataUser        string `json:"metadata,omitempty"`
+	MetadataPasswd      string `json:"metadatapasswd,omitempty"`
+	CoredataUser        string `json:"coredata,omitempty"`
+	CoredataPasswd      string `json:"coredatapasswd,omitempty"`
+	RulesengineUser     string `json:"rulesengine,omitempty"`
+	RulesenginePasswd   string `json:"rulesenginepasswd,omitempty"`
+	NotificationsUser   string `json:"notifications,omitempty"`
+	NotificationsPasswd string `json:"notificationspasswd,omitempty"`
+	SchedulerUser       string `json:"scheduler,omitempty"`
+	SchedulerPasswd     string `json:"schedulerpasswd,omitempty"`
+	LoggingUser         string `json:"logging,omitempty"`
+	LoggingPasswd       string `json:"loggingpasswd,omitempty"`
+}
+
```

Comment:
This struct will need to be touched for every service added to EdgeX. Can it be made more flexible? For example, app-functions-sdk-go will have persistence in Fuji, so some support for that will need to be added.

Reply:
I have to agree with Trevor on this, this is not scalable and needs to be changed I think